Upstream ExtensionQuery I/O error on POST request

I am trying to host openvsx locally with the public open-vsx.org repo set as its upstream repo.
I have stripped down the docker-compose.yml file to just the postgres and the server services.
I am able to use docker-compose to startup both of those services. 
I'm able to hit the swagger UI and see the getVersion call works via that UI.
I'm able to use curl to search for extensions and it returns them from upstream.
`curl -X 'GET' 'http://0.0.0.0:8080/api/-/search?query=java&targetPlatform=linux-arm64&size=5&offset=0&includeAllVersions=false' -H 'accept: application/json'`

The problem is that when I try to use the extensionquery (which is what code-server uses), I get a I/O error on the POST Request.
Here's my curl command:
`curl -X 'POST' "http://0.0.0.0:8080/vscode/gallery/extensionquery" -H "accept: application/json" -H "Content-Type: application/json" --data '{"filters":[{"criteria":[{"filterType":10,"value":"java"},{"filterType":8,"value":"Microsoft.VisualStudio.Code"},{"filterType":12,"value":"4096"}],"pageNumber":1,"pageSize":50,"sortBy":0,"sortOrder":0}],"assetTypes":[],"flags":950}' -v`

Here's parts of the stack trace:
org.springframework.web.client.ResourceAccessException: I/O error on POST request for "https://open-vsx.org/vscode/gallery/extensionquery": Read timed out
	at org.springframework.web.client.RestTemplate.createResourceAccessException(RestTemplate.java:926) ~[spring-web-6.2.15.jar:6.2.15]
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:906) ~[spring-web-6.2.15.jar:6.2.15]
	at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:741) ~[spring-web-6.2.15.jar:6.2.15]
	at org.eclipse.openvsx.adapter.UpstreamVSCodeService.extensionQuery(UpstreamVSCodeService.java:73) ~[main/:na]
	at org.eclipse.openvsx.adapter.DefaultExtensionQueryRequestHandler.getResult(DefaultExtensionQueryRequestHandler.java:40) ~[main/:na]
	at org.eclipse.openvsx.adapter.VSCodeAPI.extensionQuery(VSCodeAPI.java:94) ~[main/:na]

and lower down..

Caused by: java.net.SocketTimeoutException: Read timed out

If I replace the "http://0.0.0.0:8080/" with "https://open-vsx.org/" that curl command is able to hit the extensionquery successfully from my local terminal and also from the terminal of the VM that I am running openvsx on.

--- a/server/src/main/java/org/eclipse/openvsx/util/HttpHeadersUtil.java
+++ b/server/src/main/java/org/eclipse/openvsx/util/HttpHeadersUtil.java
@@ -22,9 +22,9 @@ public class HttpHeadersUtil {
                 var header = it.nextElement();
                 headers.add(header, request.getHeader(header));
             }
-
         } catch (IllegalStateException _) {}
         headers.remove(HttpHeaders.HOST);
+        headers.remove(HttpHeaders.CONTENT_LENGTH);
         return headers;
     }
 

